### PR TITLE
Add transaction selector based on min priority fee parameter

### PR DESCRIPTION
**Pull request by Gabriel-Trintinalia**
## PR description
Add transaction selector based on min priority fee parameter

### CODE REVIEW

**Review by fab-10 — COMMENTED** [review]
LGTM, just the CHANGELOG entry is missing that also covers the new feature added in #6080

File: ethereum/blockcreation/src/test/java/org/hyperledger/besu/ethereum/blockcreation/AbstractBlockTransactionSelectorTest.java [hunk 1]
```diff
@@ -737,6 +737,37 @@ public void transactionWithIncorrectNonceRemainsInPoolAndNotSelected() {
...
+    ProcessableBlockHeader blockHeader = createBlock(5_000_000, Wei.ONE);
+    miningParameters.setMinPriorityFeePerGas(Wei.of(7));
+    final Transaction tx1 = createTransaction(1, Wei.of(8), 100_000);
+    ensureTransactionIsValid(tx1);
+    final Transaction tx2 = createTransaction(2, Wei.of(9), 100_000);
+    ensureTransactionIsValid(tx2);
+    // transaction tx3 should not be selected
+    final Transaction tx3 = createTransaction(3, Wei.of(7), 100_000);
+    ensureTransactionIsValid(tx3);
+    transactionPool.addRemoteTransactions(List.of(tx1, tx2, tx3));
```
Comment (fab-10) [inline ethereum/blockcreation/src/test/java/org/hyperledger/besu/ethereum/blockcreation/AbstractBlockTransactionSelectorTest.java:751]: Could you add another test with EIP1559 txs?

File: ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/txselection/selectors/MinPriorityFeePerGasTransactionSelector.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+   * @param transactionSelectionResults The results of other transaction evaluations in the same
+   *     block.
+   * @return TransactionSelectionResult. If the priority fee is below the minimum, it returns an
+   *     invalid transient result. Otherwise, it returns a selected result.
+   */
+  @Override
+  public TransactionSelectionResult evaluateTransactionPreProcessing(
+      final PendingTransaction pendingTransaction,
+      final TransactionSelectionResults transactionSelectionResults) {
+    if (isPriorityFeePriceBelowMinimum(pendingTransaction.getTransaction())) {
```
Comment (fab-10) [inline ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/txselection/selectors/MinPriorityFeePerGasTransactionSelector.java:50]: as we do for minGasPrice in `PriceTransactionSelector` if a tx has priority should be accepted

File: ethereum/blockcreation/src/test/java/org/hyperledger/besu/ethereum/blockcreation/AbstractBlockTransactionSelectorTest.java (hunk 1 above)
Comment (Gabriel-Trintinalia) [reply ethereum/blockcreation/src/test/java/org/hyperledger/besu/ethereum/blockcreation/AbstractBlockTransactionSelectorTest.java]: done

File: ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/txselection/selectors/MinPriorityFeePerGasTransactionSelector.java (hunk 2 above)
Comment (Gabriel-Trintinalia) [reply ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/txselection/selectors/MinPriorityFeePerGasTransactionSelector.java]: done

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,8 @@
 - Accept `input` and `data` field for the payload of transaction-related RPC methods [#6094](https://github.com/hyperledger/besu/pull/6094)
 - Add APIs to set and get the min gas price a transaction must pay for being selected during block creation [#6097](https://github.com/hyperledger/besu/pull/6097)
 - TraceService: return results for transactions in block [#6086](https://github.com/hyperledger/besu/pull/6086)
+- New option `--min-priority-fee` that sets the minimum priority fee a transaction must meet to be selected for a block. [#6080](https://github.com/hyperledger/besu/pull/6080) [#6083](https://github.com/hyperledger/besu/pull/6083)
+- Implement new `miner_setMinPriorityFee` and `miner_getMinPriorityFee` RPC methods [#6080](https://github.com/hyperledger/besu/pull/6080)
```
Comment (macfarla) [inline CHANGELOG.md:16]: are these really related to mining?

File: ethereum/blockcreation/src/test/java/org/hyperledger/besu/ethereum/blockcreation/AbstractBlockTransactionSelectorTest.java [hunk 4]
```diff
@@ -823,6 +823,34 @@ public void decreaseOfMinGasPriceAtRuntimeIncludeTxThatWasPreviouslyNotSelected(
...
   }
 
+  @Test
+  public void shouldNotSelectTransactionsWithPriorityFeeLessThanConfig() {
+    ProcessableBlockHeader blockHeader = createBlock(5_000_000, Wei.ONE);
+    miningParameters.setMinPriorityFeePerGas(Wei.of(7));
+    final Transaction tx1 = createTransaction(1, Wei.of(8), 100_000);
+    ensureTransactionIsValid(tx1);
+    // transaction tx2 should not be selected
+    final Transaction tx2 = createTransaction(2, Wei.of(7), 100_000);
```
Comment (macfarla) [inline ethereum/blockcreation/src/test/java/org/hyperledger/besu/ethereum/blockcreation/AbstractBlockTransactionSelectorTest.java:833]: you could rename tx2 `txNotSelected`

File: ethereum/blockcreation/src/test/java/org/hyperledger/besu/ethereum/blockcreation/MinPriorityFeePerGasTransactionSelectorTest.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  public void shouldSelectWhenPrioritySender() {
+    var prioritySenderTransaction = mockTransactionWithPriorityFee(minPriorityFeeParameter - 1);
+    assertSelection(
+        prioritySenderTransaction,
+        TransactionSelectionResult.PRIORITY_FEE_PER_GAS_BELOW_CURRENT_MIN);
+    when(prioritySenderTransaction.hasPriority()).thenReturn(true);
+    assertSelection(prioritySenderTransaction, TransactionSelectionResult.SELECTED);
+  }
+
+  private void assertSelection(
```
Comment (macfarla) [inline ethereum/blockcreation/src/test/java/org/hyperledger/besu/ethereum/blockcreation/MinPriorityFeePerGasTransactionSelectorTest.java:85]: maybe `assertSelectionResult` is a better name?

**Review by macfarla — APPROVED** [review]
some naming nits

**Review by fab-10 — APPROVED** [review]

File: ethereum/blockcreation/src/test/java/org/hyperledger/besu/ethereum/blockcreation/MinPriorityFeePerGasTransactionSelectorTest.java (hunk 5 above)
Comment (Gabriel-Trintinalia) [reply ethereum/blockcreation/src/test/java/org/hyperledger/besu/ethereum/blockcreation/MinPriorityFeePerGasTransactionSelectorTest.java]: done

File: CHANGELOG.md (hunk 3 above)
Comment (Gabriel-Trintinalia) [reply CHANGELOG.md:16]: they are related to block creation. I think it is the same case of this: https://github.com/hyperledger/besu/pull/6027#discussion_r1368620223

Comment (fab-10) [reply CHANGELOG.md:16]: the rename from _miner_ to _block creation_ is on the todo list, but it is not trivial and should be done with care, it will introduce a lot of breaking changes and need for external tools to update. 
I suggest that for the moment we keep it as is, and we consider _miner_ a synonym of _block creation_
